### ISSUE TITLE
QueryPurchases returns purchases for all accounts in device, queryPurchaseHistoryAsync doesn't

### ISSUE THREAD

**Issue by vegaro** [android/play-billing-samples#318]
If there are multiple accounts added to the Play Store, `queryPurchases` returns the active purchases of all accounts logged in the device. Which means that if two accounts are configured, and both have the same subscription, `queryPurchases` will return two items.

On the other hand, `queryPurchaseHistoryAsync` only returns the last purchase for **one** of the logged-in accounts. I believe it would return the history of what is consider the original account (the first one added to the phone).

I am upgrading to 3.0.0 and `setOldSku` requieres a `purchaseToken` to disambiguate situations where multiple accounts on the device own the same sku.

How are we supposed to disambiguate on our side and choose which `purchaseToken` to send to `setOldSku` if `queryPurchases` returns all purchase tokens? I could ask my backend which `purchaseToken` is associated with my app user, but what if I am calling `queryPurchases` to actually send my backend the `purchaseToken` and determine if the user should have access to my app? Which `purchaseToken` would I send to my backend to associate with my app user?

**Comment by ashughes** [comment]
`queryPurchases` does not return the active purchases of all accounts on the device. This is something I've been advocating for for many years. Please see the discussion at https://github.com/android/play-billing-samples/issues/2 and this bug on the official Google issue tracker: https://issuetracker.google.com/issues/139597485.

Are you seeing `queryPurchases` return active purchases from multiple accounts? Or was this just your understanding/assumption of how `queryPurchases` works and you were trying to figure out how to handle the purchases if this were to occur?

**Comment by vegaro** [comment]
I tested this specifically the day I opened the issue and saw `queryPurchases` return both purchases for both accounts. I found a partial screenshot I took that day of the contents of the list inside the PurchasesResult:

![Screen Shot 2020-06-23 at 8 01 08 PM](https://user-images.githubusercontent.com/664544/86864964-f7915600-c082-11ea-9116-da120547c787.png)

The problem is... I can't reproduce it anymore 😢  What I did was:

1. Purchased a monthly subscription with user A (5 minutes duration in a testing environment).
3. Logout user A in the Play Store
4. Login user B in the Play Store
5. Purchased a monthly subscription with user B of the same monthly subscription
7. Login user A again. Both user A and B are logged in the Play Store
8. QueryPurchases returns both purchases inside the PurchasesResult

I will keep trying to reproduce it, just in case I see it again, but I will close this issue to not add noise.
